pkg/diff/diff.go

```diff
@@ -3,14 +3,32 @@ package diff
 import (
 	"bufio"
 	"bytes"
+	"errors"
 	"fmt"
 	"os/exec"
+	"strings"
 
 	"github.com/go-git/go-billy/v5"
 	"github.com/rancher/charts-build-scripts/pkg/filesystem"
 	"github.com/sirupsen/logrus"
 )
 
+func checkDependencyPackage(pathToCmd string) error {
+	cmd := exec.Command(pathToCmd, "--version")
+	out, err := cmd.Output()
+	if err != nil {
+		return err
+	}
+
+	version := string(out)
+
+	if strings.Contains(version, "Apple") || strings.Contains(version, "FreeBSD") {
+		return errors.New("detected Apple/FreeBSD version of patch. This will lead to compatibility issues. Install GNU patch: https://github.com/rancher/charts-build-scripts/issues/130")
```

Comment:
As this check is not specific to `patch` command, should we modify the error message something similar to? 
```suggestion
		return errors.Errorf("detected Apple/FreeBSD version of %[1]s. This will lead to compatibility issues. Install GNU %[1]s: https://github.com/rancher/charts-build-scripts/issues/130", pathToCmd)
```